File: Source/UserManagement/Domain/DataCollector/Registering/RegisterDataCollector.cs
/*---------------------------------------------------------------------------------------------
 *  Copyright (c) 2017 International Federation of Red Cross. All rights reserved.
 *  Licensed under the MIT License. See LICENSE in the project root for license information.
 *--------------------------------------------------------------------------------------------*/

using System;
using System.Collections.Generic;
using Concepts;
using Dolittle.Commands;

namespace Domain.DataCollector.Registering
{
    public class RegisterDataCollector : ICommand
    {
        public Guid DataCollectorId { get; set; }

        
        public string FullName { get; set; }
        public string DisplayName { get; set; }
        public int YearOfBirth { get; set; }
        public Sex Sex { get; set; }
        public Language PreferredLanguage { get; set; }
        public Location GpsLocation { get; set; }
        public IEnumerable<string> PhoneNumbers { get; set; }
        
        // Todo: we can deduct this from the command handler
        public DateTimeOffset RegisteredAt { get; set; }

        // Todo: This is wrong - the fact that it is a Register command means it is New by definition; implicitly
        public bool IsNewRegistration { get; set; }
    }
}

File: Source/UserManagement/Web.Angular/src/app/domain/RegisterDataCollector.ts
import {  Command } from '../services/Command';
import { Location } from './location.model';
import { Sex } from './sex';
import { Language } from './language.model';

export class RegisterDataCollector extends Command {
    dataCollectorId: string;
    fullName: string;
    displayName: string;
    yearOfBirth: number;
    sex: Sex;
    preferredLanguage: Language;
    gpsLocation: Location;
    phoneNumbers: Array<string>;

    constructor() {
        super();
        this.type = 'CBS#VolunteerReporting.DataCollector-RegisterDataCollector+Command|Domain';
        this.gpsLocation = new Location();
    }
}
